SMLHelper/Patchers/AtmosphereVolumePatcher.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HarmonyLib;
+using UnityEngine;
+using System.Reflection.Emit;
+namespace SMLHelper.V2.Patchers
+{
+    internal class AtmosphereVolumePatcher
+    {
+            [HarmonyPatch(typeof(AtmosphereVolume), nameof(AtmosphereVolume.Start))]
+            [PatchUtils.Transpiler]
+            internal static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> list_)
+            {
+                var list = new List<CodeInstruction>(list_);
+                for(var i = 0; i < list.Count;i++)
+                {
+                    if (list[i].Calls(typeof(Component).GetMethod(nameof(Component.GetComponent),Type.EmptyTypes).MakeGenericMethod(typeof(Collider))))
+                    {
+                        list[i].operand = typeof(AtmosphereVolumePatcher).GetMethod(nameof(AtmosphereVolumePatcher.GetComponentReplace),System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Static);
+                    }
+                }
+                return list.AsEnumerable();
+            }
+            internal static unsafe Collider GetComponentReplace(Component __instance)
```

Comment:
What does this patch do? Seems unnecessary.

Reply:
The patch makes it so that the Collider for the AtmosphereVolume and the collider for the terrain isn't confused by the game using GetComponent<Collider>()